jupyterhub/nullauthenticator.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+from tornado import web
+
+from .auth import Authenticator
+from .handlers.base import BaseHandler
+
+
+class NullLoginHandler(BaseHandler):
+    def get(self):
+        raise web.HTTPError(403, "Login is not supported")
+
+
+class NullAuthenticator(Authenticator):
+    """Null Authenticator for JupyterHub
+
+    For cases where authentication should be disabled,
+    e.g. only allowing access via API tokens.
+
+    .. versionadded:: 2.0
+    """
+
+    # auto_login skips 'Login with...' page on Hub 0.8
+    auto_login = True
+
+    # for Hub 0.7, show 'login with...'
+    login_service = 'null'
+
+    def get_handlers(self, app):
+        return [('/nologin', NullLoginHandler)]
```

Comment:
What's the purpose of this `/nologin` handler? The original https://github.com/jupyterhub/nullauthenticator/blob/dedb18412a06da9e309c1b92c05dd305ec817d03/nullauthenticator.py#L18 uses the `web` module which isn't included in the imports, so `NullLoginHandler.get()` will return a 500 server error instead of the intended 403.

I could imagine it being used as the `login_url()` https://github.com/jupyterhub/jupyterhub/blob/b029d983f99878b365c28a6e8b606cfeb0e8d3d7/jupyterhub/auth.py#L673-L688
but that would require overriding it.

Reply:
I think this is a leftover from Authenticators maybe *needing* to register a handler to be redirected in earlier versions of JupyterHUb.

I think `return []` is appropriate here. The Handler itself is not needed anymore.

*alternately*, this could be `/login` and forcefully override the default login page.

The result should ultimately be the same, though, so the simplest option of an empty list and no custom handler is probably correct.